fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/impl/UpdateRDFSourcePersisterTest.java

```diff
@@ -0,0 +1,188 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.InputStream;
+import java.time.Instant;
+import java.util.stream.Stream;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.Triple;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFDataMgr;
+import org.apache.jena.vocabulary.DC;
+import org.fcrepo.kernel.api.RdfStream;
+import org.fcrepo.kernel.api.models.ResourceHeaders;
+import org.fcrepo.kernel.api.operations.RdfSourceOperation;
+import org.fcrepo.kernel.api.rdf.DefaultRdfStream;
+import org.fcrepo.persistence.api.WriteOutcome;
+import org.fcrepo.persistence.ocfl.api.FedoraToOCFLObjectIndex;
+import org.fcrepo.persistence.ocfl.api.OCFLObjectSession;
+import org.fcrepo.persistence.ocfl.api.OCFLObjectSessionFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * @author dbernstein
+ * @since 6.0.0
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class UpdateRDFSourcePersisterTest {
+
+    private static final String RESOURCE_ID = "info:fedora/parent/child";
+
+    private static final String ROOT_RESOURCE_ID = "info:fedora/parent";
+
+    private static final String USER_PRINCIPAL = "fedoraUser";
+
+    private static final Instant CREATED_DATE = Instant.parse("2019-11-22T12:10:04.000000Z");
+
+    private static final Instant MODIFIED_DATE = Instant.parse("2019-11-22T12:40:33.697004Z");
+
+    private static final String TITLE = "My title";
+
+    @Mock
+    private RdfSourceOperation operation;
+
+    @Mock
+    private OCFLObjectSession session;
+
+    @Mock
+    private FedoraOCFLMapping mapping;
+
+    @Mock
+    private FedoraToOCFLObjectIndex index;
+
+    @Mock
+    private OCFLObjectSessionFactory sessionFactory;
+
+    @Mock
+    private OCFLPersistentStorageSession psSession;
+
+    @Mock
+    private WriteOutcome writeOutcome;
+
+    @Captor
+    private ArgumentCaptor<InputStream> userTriplesIsCaptor;
+
+    @Captor
+    private ArgumentCaptor<InputStream> headersIsCaptor;
+
+    private UpdateRDFSourcePersister persister;
+
+    @Before
+    public void setup() throws Exception {
+        operation = mock(RdfSourceOperation.class);
+
+        when(session.write(anyString(), any(InputStream.class))).thenReturn(writeOutcome);
+        when(psSession.findOrCreateSession(anyString())).thenReturn(session);
+        when(index.getMapping(anyString())).thenReturn(mapping);
+
+        persister = new UpdateRDFSourcePersister(this.sessionFactory, this.index );
+    }
+
+    @Test
+    public void testPersistExistingResource() throws Exception {
```

Comment:
test for updating a resource that doesn't exist

Reply:
I know that the AbstractPersisterTest tests `handle` in a generic sense, but it would probably be good to verify that the persister impls handle the expected types of operations, since that is unique to each persister.

Reply:
This persister should never be called when the resource does not exist.  That case will be handled by CreateRDFSourcePersister.

Reply:
We'll check at a higher level for updating resources that don't exist.